File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    answer = 1
    m = 1
    
    for i in range(1, a+1):
        if a % i == 0 and b % i == 0:
            m = i
    b //= m
    
    while b % 2 == 0:
        b /= 2
            
    while b % 5 == 0:
        b /= 5
        
    if b != 1:
        answer = 2
    
    return answer

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        int answer = 1;
        int max = 1;
        
        for (int i = 1; i <= a; i++){
            if (a % i == 0 && b % i == 0){
                max = i;
            }
        }
        
        b /= max;
        
        while (b % 2 == 0) {
            b /= 2;
        }
        while (b % 5 == 0) {
            b /= 5;
        }
        
        if (b != 1){
            answer = 2;
        }
        
        return answer;
    }
}